clients/src/test/java/org/apache/kafka/clients/admin/MockAdminClient.java

```diff
@@ -238,7 +238,7 @@ synchronized public CreateTopicsResult createTopics(Collection<NewTopic> newTopi
                 createTopicResult.put(topicName, future);
                 continue;
             }
-            int replicationFactor = newTopic.replicationFactor();
+            int replicationFactor = Math.max(newTopic.replicationFactor(), 0);
```

Comment:
By default, the partitions and replications are set to default value from broker's config when they are ignored. This PR sets the default value to 0 and it causes topic metadata contains zero replica and partition.
```
            List<Node> replicas = new ArrayList<>(replicationFactor);
            for (int i = 0; i < replicationFactor; ++i) {
                replicas.add(brokers.get(i));
            }
```

I worry that the following operations, like #describeTopics, will return incorrect information about topic.

Reply:
You're right! I overlooked this. I'll review the PR. Thanks